### PR TITLE
[fix] refactor db scanning code to be more resilient

### LINKED ISSUE #161: refactor to fix how we scan results
Long story short, we should switch from calling row.Scan to sqlx.StructScan everywhere.

The former approach is brittle in the face of new columns being added (or re-ordered).

### PR DESCRIPTION
**Pull request by ryanking**
Scan all queries with sqlx.StructScan in unsafe mode so that we are resilient to snowflake adding new columns.

Because we have been using sql.Row.Scan, which requires that you list variables for each column you want to pull out (in order), if snowflake added a new column to any of our queries, code would break.

## Test Plan
* [x] acceptance tests

## References
* Fixes #161

### CODE REVIEW

File: pkg/snowflake/stage.go [hunk 1]
```diff
@@ -190,3 +193,76 @@ func (sb *StageBuilder) Describe() string {
...
+	AwsExternalID    string
+	SnowflakeIamUser string
+	FileFormat       string
+	CopyOptions      string
+}
+
+type descStageRow struct {
+	parentProperty  string `db:"parent_property"`
+	property        string `db:"property"`
+	propertyType    string `db:"property_type"`
```
Comment (cziie) [inline pkg/snowflake/stage.go:222]: **[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
`propertyType` is unused (structcheck)

File: pkg/snowflake/exec.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+func Exec(db *sql.DB, query string) error {
+	log.Print("[DEBUG] stmt ", query)
+
+	_, err := db.Exec(query)
+	return err
+}
+
+// QueryRow will run stmt against the db and return the row. We use
+// [DB.Unsafe](https://godoc.org/github.com/jmoiron/sqlx#DB.Unsafe) so that we can scan to structs
```
Comment (edulop91) [inline pkg/snowflake/exec.go:18]: wish this wasn't named unsafe

File: pkg/snowflake/role.go [hunk 3]
```diff
@@ -1,8 +1,25 @@
...
+
+type role struct {
+	Name    sql.NullString `db:"name"`
+	Comment sql.NullString `db:"comment"`
+}
+
+func ScanRole(row *sqlx.Row) (*role, error) {
+	r := &role{}
+	e := row.StructScan(r)
+	return r, e
```
Comment (edulop91) [inline pkg/snowflake/role.go:24]:
```suggestion
	err := row.StructScan(r)
	return r, err
```
for consistency/convention

File: pkg/snowflake/schema.go [hunk 4]
```diff
@@ -172,3 +175,17 @@ func (sb *SchemaBuilder) Show() string {
...
+	DatabaseName  sql.NullString `db:"database_name"`
+	Comment       sql.NullString `db:"comment"`
+	Options       sql.NullString `db:"options"`
+	RetentionTime sql.NullInt64  `db:"retention_time"`
+}
+
+func ScanSchema(row *sqlx.Row) (*schema, error) {
+	r := &schema{}
+	e := row.StructScan(r)
+	return r, e
```
Comment (edulop91) [inline pkg/snowflake/schema.go:190]:
```suggestion
	err := row.StructScan(r)
	return r, err
```

File: pkg/snowflake/share.go [hunk 5]
```diff
@@ -16,3 +22,15 @@ func Share(name string) *Builder {
...
+type share struct {
+	Name    sql.NullString `db:"name"`
+	To      sql.NullString `db:"to"`
+	Comment sql.NullString `db:"comment"`
+}
+
+func ScanShare(row *sqlx.Row) (*share, error) {
+	r := &share{}
+	e := row.StructScan(r)
+	return r, e
```
Comment (edulop91) [inline pkg/snowflake/share.go:35]:
```suggestion
	err := row.StructScan(r)
	return r, err
```

File: pkg/snowflake/stage.go [hunk 6]
```diff
@@ -190,3 +193,75 @@ func (sb *StageBuilder) Describe() string {
...
+	DatabaseName       *string `db:"database_name"`
+	SchemaName         *string `db:"schema_name"`
+	Comment            *string `db:"comment"`
+	StorageIntegration *string `db:"storage_integration"`
+}
+
+func ScanStageShow(row *sqlx.Row) (*stage, error) {
+	r := &stage{}
+	e := row.StructScan(r)
+	return r, e
```
Comment (edulop91) [inline pkg/snowflake/stage.go:208]:
```suggestion
	err := row.StructScan(r)
	return r, err
```

File: pkg/snowflake/storage_integration.go [hunk 7]
```diff
@@ -16,3 +22,17 @@ func StorageIntegration(name string) *Builder {
...
+	Category        sql.NullString `db:"category"`
+	IntegrationType sql.NullString `db:"integration_type"`
+	CreatedOn       sql.NullString `db:"created_on"`
+	Enabled         sql.NullBool   `db:"enabled"`
+}
+
+func ScanStorageIntegration(row *sqlx.Row) (*storageIntegration, error) {
+	r := &storageIntegration{}
+	e := row.StructScan(r)
+	return r, e
```
Comment (edulop91) [inline pkg/snowflake/storage_integration.go:37]:
```suggestion
	err := row.StructScan(r)
	return r, err
```

File: pkg/snowflake/stage.go [hunk 8]
```diff
@@ -190,3 +193,75 @@ func (sb *StageBuilder) Describe() string {
...
+}
+
+type descStageRow struct {
+	parentProperty  string `db:"parent_property"`
+	property        string `db:"property"`
+	propertyValue   string `db:"property_value"`
+	propertyDefault string `db:"property_default"`
+}
+
+func DescStage(db *sql.DB, query string) (descStageResult, error) {
```
Comment (edulop91) [inline pkg/snowflake/stage.go:226]:
```suggestion
func DescStage(db *sql.DB, query string) (*descStageResult, error) {
```
Might make things clearer, so you don't have to return the empty descStageRow + for consistency with the way the other functions work

File: pkg/snowflake/user.go [hunk 9]
```diff
@@ -1,8 +1,31 @@
...
+	Disabled         bool           `db:"disabled"`
+	HasRsaPublicKey  bool           `db:"has_rsa_public_key"`
+	LoginName        sql.NullString `db:"login_name"`
+	Name             sql.NullString `db:"name"`
+}
+
+func ScanUser(row *sqlx.Row) (*user, error) {
+	r := &user{}
+	e := row.StructScan(r)
+	return r, e
```
Comment (edulop91) [inline pkg/snowflake/user.go:30]:
```suggestion
	err := row.StructScan(r)
	return r, err
```

File: pkg/snowflake/view.go [hunk 10]
```diff
@@ -146,3 +149,18 @@ func (vb *ViewBuilder) Show() string {
...
+	Name         sql.NullString `db:"name"`
+	SchemaName   sql.NullString `db:"schema_name"`
+	Text         sql.NullString `db:"text"`
+	DatabaseName sql.NullString `db:"database_name"`
+}
+
+func ScanView(row *sqlx.Row) (*view, error) {
+	r := &view{}
+	e := row.StructScan(r)
+	return r, e
```
Comment (edulop91) [inline pkg/snowflake/view.go:165]:
```suggestion
	err := row.StructScan(r)
	return r, err
```

File: pkg/snowflake/warehouse.go [hunk 11]
```diff
@@ -1,8 +1,54 @@
...
+	Failed          int64     `db:"failed"`
+	Suspended       int64     `db:"suspended"`
+	UUID            string    `db:"uuid"`
+	ScalingPolicy   string    `db:"scaling_policy"`
+}
+
+func ScanWarehouse(row *sqlx.Row) (*warehouse, error) {
+	w := &warehouse{}
+	e := row.StructScan(w)
+	return w, e
```
Comment (edulop91) [inline pkg/snowflake/warehouse.go:53]:
```suggestion
	err := row.StructScan(w)
	return w, err
```

**Pull request merged**